kodilla-jdbc/src/test/java/com/kodilla/jdbc/DbManagerTestSuite.java

```diff
@@ -31,11 +31,37 @@ public void testSelectUsers() throws SQLException {
         while(rs.next()) {
             System.out.println(rs.getInt("ID") + ", " +
                     rs.getString("FIRSTNAME") + ", " +
-                    rs.getString("LASTNAM"));
+                    rs.getString("LASTNAME"));
             counter++;
         }
         rs.close();
         statement.close();
         Assert.assertEquals(5, counter);
     }
+
+    @Test
+    public void testSelectUsersAndPosts() throws SQLException {
+        //Given
+        DbManager dbManager = DbManager.getInstance();
+        //When
+        String sqlQuery = "SELECT FIRSTNAME, LASTNAME, COUNT(*) AS no_of_posts\n" +
+                "FROM USERS\n" +
+                "JOIN POSTS\n" +
+                "ON USERS.ID = POSTS.USER_ID\n" +
+                "GROUP BY USER_ID\n" +
+                "HAVING COUNT(*) > 1";
+        Statement statement = dbManager.getConnection().createStatement();
+        ResultSet rs = statement.executeQuery(sqlQuery);
+        //Then
+        int counter = 0;
+        while(rs.next()) {
+            System.out.println(rs.getString("FIRSTNAME") + ", " +
+                    rs.getString("LASTNAME") + ", "+
+                    rs.getInt("no_of_posts"));
+            counter++;
+        }
+        rs.close();
+        statement.close();
+        Assert.assertEquals(2, counter);
```

Comment:
Dla czystości testu sekcja cleanup (`rs.close()` i `statement.close()`) powinna być pod then. Z drugiej strony, jeśli asercja nie przejdzie, zostaniesz z niezamkniętym połączeniem, więc sytuacja jest trochę patowa. To wynika trochę z tego, że normalnie takich rzeczy nie pisze się w testach, no ale tutaj było takie zadanie :)